numpy/lib/financial.py

```diff
@@ -645,7 +645,11 @@ def rate(nper, pmt, pv, fv, when='end', guess=None, tol=None, maxiter=100):
         rn = rnp1
     if not close:
         # Return nan's in array of the same shape as rn
-        return np.nan + rn
+        if type(rn)==Decimal:
+            # Type conversion of np.nan to class decimal.Decimal
+            return Decimal(np.nan) + rn
```

Comment:
I think you can just use `defaullt_type` here

Reply:
Hi Eric,
Thanks for the review.  Sorry,I didnt get "default_type". You mean to say just return np.nan + rn as it is?
My concern is, in case of returning np.nan + rn as it is, without type conversion, type error is coming in case of below scenario:
np.rate(Decimal(12), Decimal('400'), Decimal('10000'), Decimal(0))

rn variable type is Decimal in above case, which is not compatible with np.nan and thus throwing type error.

Please correct me if i am wrong (I'm newbie to open source contribution but really want to contribute to it)

Regards,
Mandeep Singh

Reply:
I mean drop the `if`, and replace `np.nan` with `default_type(np.nan)`

Reply:
Thanks Eric for the help :) . I will change it and add a new pull request